### ISSUE TITLE
Upgraded rooms disappear in space

### ISSUE THREAD

**Issue by yostyle** [vector-im/element-android#4866]
### Steps to reproduce

On web client
1. Create a room
2. Create a space
3. Add the previous room to the space
4. Upgrade the room

On Android
1. Clear the cache or install the app
2. Go to the space

#### What did you expect?
The upgraded room is not present in the space on Android client.

#### What happened instead?
The upgraded room should be present.

### Your phone model

xiaomi Mi 9

### Operating system version

Android 10

### Application version and app store

1.3.12

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by bmarty** [comment]
After a quick investigation, the state event in the space still reference the old roomId.

**Comment by kittykat** [comment]
> After a quick investigation, the state event in the space still reference the old roomId.

Is this an Android issue or an upstream Spaces issues? Gil demoed it earlier today working properly on iOS

**Comment by golanv** [comment]
I upgraded several rooms with the Linux (Electron) client.  One of the rooms disappeared from the space room list.  New users to the space cannot see the room.  I don't think that's very helpful, but the issue I'm seeing appears to be similar and not limited to Android.

**Comment by jivanpal** [comment]
@golanv Yes, that seems like a separate issue since it's affecting all clients, so probably an issue on the part of your homeserver.

I'm just chiming in here to say that I'm also experiencing this in Element Android, and have been since before the introduction of the new UI. Element Web/Desktop handles this correctly (upgraded rooms show up there in the space that the old room is in), but I haven't looked through the Spaces spec to see which is the intended behaviour as far as the spec is concerned. It makes sense from a UX perspective to follow the room ID references to see if the room has been upgraded and determine the room ID of the new room, like Element Web does.

Homeserver: opaline.uk (endpoint `matrix.opaline.uk/_matrix`), Synapse 1.69.0
Affected client: Element Android 1.5.2 (Google Play)
